### PR TITLE
Add a workflow to separate out eslint and annotate PRs.

### LINKED ISSUE #33758: [Enhancement]: Show ESLint issues visually in PR diffs
### Describe the solution you'd like

Various types of PHPCS issues are flagged visually:

example-phpcs-issue

At least for legacy JS, we don't seem to have the same thing ... but it could be handy. 

### Describe alternatives you've considered

_No response_

### Additional context

See also https://github.com/woocommerce/woocommerce/issues/33757: at present, an ESLint error can stop unit tests from running—so it would be particularly valuable to surface them (presently, they are buried deep in the action output).

### PR DESCRIPTION
**Pull request by samueljseay**
## Changes

This PR separates out the JS testing checks from the linting checks and it annotates eslint errors in the PR.

It closes #33758 and #33757

## Testing

To test this, I've tested it live in the PR by introducing an eslint failure, you can see that the PR check continues to run after lint failure and then annotates the file (have a look at the file in the PR).

Once this is approved and ready to merge I'll remove the linted change.

I don't think there is much else to test here. I've tested the success case too manually in the PR.

**There is just one thing left to do before merging this: we'll need to disable the required check for the old combined job of lint and test and make these new separate jobs of test and lint required.** I will do this before merging.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Make eslint emit JSON report for annotating PRs.

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/pr-lint-js.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+            - name: Lint
+              run: pnpm run -r --filter='release-posts' --filter='woocommerce/client/admin...' --filter='@woocommerce/monorepo-utils' --filter='!@woocommerce/e2e*' --filter='!@woocommerce/api' --color lint
+              continue-on-error: true
+
+            - name: Collect and Combine Eslint Reports
+              run: node ./.github/workflows/scripts/collect-eslint-reports.js
+
+            - name: Annotate Code Linting Results
+              uses: ataylorme/eslint-annotate-action@v2
```
Comment (jonathansadowski) [inline .github/workflows/pr-lint-js.yml]: Should we pin this to a specific hash?

**Review by jonathansadowski — APPROVED** [review]
Looks good, nice work! I asked a question about pinning one of the actions to a specific hash, but aside from that, 👍

Feel free to merge once addressing comment and removing the linting issue.

File: .github/workflows/pr-lint-js.yml (hunk 1 above)
Comment (samueljseay) [reply .github/workflows/pr-lint-js.yml]: Good call, we should always do that. Thanks for spotting!

Comment (samueljseay) [reply .github/workflows/pr-lint-js.yml]: Resolved in https://github.com/woocommerce/woocommerce/pull/39704/commits/8a197ef967d4ce0346fdc4a3b19f7597de7ad428

**Pull request merged**